backend/src/main/kotlin/be/osoc/team1/backend/controllers/ControllersUtil.kt

```diff
@@ -19,11 +32,67 @@ import org.springframework.web.servlet.support.ServletUriComponentsBuilder
  *
  * `/api/students/(INSERT ID)`
  */
-fun <ID, T> getObjectCreatedResponse(id: ID, createdObject: T, status: HttpStatus = HttpStatus.CREATED): ResponseEntity<T> {
+fun <ID, T> getObjectCreatedResponse(
+    id: ID,
+    createdObject: T,
+    status: HttpStatus = HttpStatus.CREATED
+): ResponseEntity<T> {
     val postRequestPath = ServletUriComponentsBuilder.fromCurrentRequest()
     val pathWithIdAdded = postRequestPath.path("/{id}").buildAndExpand(id).toUriString()
     return ResponseEntity
         .status(status)
         .header(HttpHeaders.LOCATION, pathWithIdAdded)
         .body(createdObject)
 }
+
+/**
+ * This function checks if a request is allowed based on the edition of the accessed resource ([editionName]),
+ * (entries of inactive editions can only be viewed or deleted by admins).
+ * [editionService], [userDetailService] and [httpServletRequest] should be injected in the caller of this function.
+ */
+fun attemptEditionAccess(
+    editionName: String,
+    editionService: EditionService,
+    userDetailService: OsocUserDetailService,
+    httpServletRequest: HttpServletRequest
+) {
+    val edition = editionService.getEdition(editionName)
+    val authentication = SecurityContextHolder.getContext().authentication
+    val user = userDetailService.getUserFromPrincipal(authentication)
+
+    if (!edition.accessibleBy(user))
+        throw UnauthorizedOperationException("Entries of inactive editions can only be accessed by admins!")
+    if (!edition.isActive && httpServletRequest.method != "GET" && httpServletRequest.method != "DELETE")
+        throw ForbiddenOperationException("Entries of inactive editions can only be viewed or delete (Allowed methods: GET, DELETE)")
+}
+
+/**
+ * This class adds the code of the SecuredEdition annotation
+ */
+@Aspect
+@Component
+class EditionSecurityAspect(val editionService: EditionService, val userDetailService: OsocUserDetailService) {
+
+    @Autowired
+    private lateinit var request: HttpServletRequest
+
+    @Before(value = "@annotation(SecuredEdition)")
+    @Throws(Throwable::class)
+    fun validateEditionArgument(joinPoint: JoinPoint): Any {
+        val method = (joinPoint.signature as MethodSignature).method
+        val editionFieldIndex = method.parameters.indexOfFirst { it.name == "edition" }
+        if (editionFieldIndex < 0)
+            throw IllegalStateException("The @SecuredEdition edition argument was not found! (With this annotation the function needs an edition argument)")
+        val editionName = joinPoint.args[editionFieldIndex] as String
```

Comment:
I think this would be cleaner if you did
```kotlin
val editionField = method.parameters.firstOrNull { it.name == "edition" }
if (editionField == null)
    throw IllegalStateException("...")
```

Reply:
You can't get the value of the edition like that, firstOrNull will return a Parameter class. To get the actual value we use joinPoint.args[index] so we do need the index here.